spec/view_integration/post_spec.rb

```diff
@@ -0,0 +1,36 @@
+require 'rails_helper'
+
+RSpec.describe 'Posts Workflow', type: :feature do
+  let!(:user) { create(:user, posts_counter: 3) }
+  let!(:post) { create(:post, author: user, comments_counter: 2, likes_counter: 2) }
+
+  describe 'User post index page' do
+    it 'displays user information, profile picture, and post count' do
+      visit user_posts_path(user)
+      expect(page).to have_content(user.name)
+      expect(page).to have_selector("img[src='#{user.photo}']")
+    end
```

Comment:
- You have done well so far, but according to the project requirements, you should test for the number of posts a user has written. Please fix this 🙏